src/handlers/int-test-support/helpers/s3Helper.ts

```diff
@@ -32,20 +33,44 @@ interface DataAndTarget {
   target: S3Object;
 }
 
+interface deleteS3ObjectByKeys {
+  bucket: string;
+  keysToDelete: ObjectIdentifier[];
+}
+
+interface deleteS3ObjectByPrefixes {
+  bucketName: string;
+  prefixesToDelete: string[];
+}
+
 const listS3ObjectsBasic = async (
   params: BucketAndPrefix
-): Promise<ListObjectsCommandOutput> => {
+): Promise<ListObjectsV2CommandOutput> => {
   if (runViaLambda())
     return (await sendLambdaCommand(
       IntTestHelpers.listS3Objects,
       params
-    )) as unknown as ListObjectsCommandOutput;
+    )) as unknown as ListObjectsV2CommandOutput;
+
+  const objects: ListObjectsV2CommandOutput = { Contents: [], $metadata: {} };
+  let continuationToken: string | undefined;
+  do {
+    const listParams = {
+      Bucket: params.bucketName,
+      Prefix: params.prefix,
+      ContinuationToken: continuationToken,
+    };
+    const listResult = await s3Client.send(
+      new ListObjectsV2Command(listParams)
+    );
+    if (listResult.Contents === undefined) {
+      return objects;
+    }
+    objects.Contents = objects.Contents?.concat(listResult.Contents);
```

Comment:
We should not use the AWS datatype where we don't need it. You will never be populating the metadata bit of the type and neither should you. Just return the array of objects.

Reply:
ok will return just array of objects